crates/ra_hir/src/source_binder.rs

```diff
@@ -409,6 +412,37 @@ impl SourceAnalyzer {
         crate::ty::autoderef(db, &self.resolver, canonical).map(|canonical| canonical.value)
     }
 
+    /// Checks that particular type `ty` implements `std::future::Future`.
+    /// This function is used in `.await` syntax completion.
+    pub fn impls_future(&self, db: &impl HirDatabase, ty: Ty) -> bool {
+        let std_future_path = Path {
+            kind: PathKind::Abs,
+            segments: vec![
+                PathSegment { name: name::STD, args_and_bindings: None },
+                PathSegment { name: name::FUTURE_MOD, args_and_bindings: None },
+                PathSegment { name: name::FUTURE_TYPE, args_and_bindings: None },
+            ],
```

Comment:
(not in this pull request)

We should consolidate known items, like `std::iter::Iterator` or `std::future::Future` *somewhere*. Perhaps they make sense as methods on `Resolver`?

Reply:
Yes. I can create a PR for that.